client/layout/index.jsx

```diff
@@ -100,6 +101,19 @@ class Layout extends Component {
 		// intentionally don't remove these in unmount
 	}
 
+	renderMasterbar() {
+		const MasterbarComponent = config.isEnabled( 'jetpack-cloud' )
```

Comment:
Not a blocker at all, but the way this is written seems weird. It's crafty, yes, and I commend it. However, it's hard to understand what's going on at a glance. Prefer something like 

`return config.isEnabled( 'jetpack-cloud' )
	? <JetpackCloudMasterbar ...>
	: <MasterbarLoggedIn ...>`

Reply:
The reason why it is written this way is to emphasis the differences and similarities. 
The only difference currently is the is the component function, the props (function arguments) are the same. 

Written the other way the emphasis is on both. (component and props) 

I am happy to rewrite it, but I think this approach conveys more meaning. I am happy to change it, but in my previous review the feedback I got from @tyxla was to write it this way.

Reply:
Yup, I agree with the approach here. The only downside is that we're always importing both components, even though we're going to use just one of them, but it's a compromise we should be fine with, given that the masterbar is immediately visible and is the first visible thing at the top of the page.